.github/workflows/benchmarks.yml

```diff
@@ -10,33 +10,81 @@ on:
 env:
   CARGO_TERM_COLOR: always
   RUST_LOG: info
+  PARTITIONS: 8
 
 jobs:
-  rebuild:
+  # rebuild:
+  #   runs-on: ubuntu-latest
+  #   steps:
+  #   - uses: actions/checkout@v4
+  #   - run: rustup update
+  #   - name: Rebuild benchmarks
+  #     run: benchmarks/build-all.sh
+
+  build-wasmtime:
     runs-on: ubuntu-latest
+    strategy:
+      matrix:
+        # for now, limit this a bit
+        revision: ["main", "v25.0.0"]
+    env:
+      REVISION: ${{ matrix.revision }}
     steps:
-    - uses: actions/checkout@v2
-    - name: Install nightly
-      uses: actions-rs/toolchain@v1
+    - uses: actions/checkout@v4
+    - run: rustup update
+    - name: Build Engine
+      run: |
+        pushd engines/wasmtime
+        rustc build.rs
+        ./build
+        popd
+    - name: Upload Wasmtime Shared Library
+      uses: actions/upload-artifact@v4
       with:
-        toolchain: nightly
-    - name: Rebuild benchmarks
-      run: benchmarks/build-all.sh 5
+        name: wasmtime-${{ matrix.revision }}
+        path: engines/wasmtime/libengine.so
 
-  run:
+  benchmark:
     runs-on: ubuntu-latest
+    needs: build-wasmtime
+    strategy:
+      matrix:
+        partition: ["00", "01", "02", "03", "04", "05", "06", "07"]
     steps:
-    - uses: actions/checkout@v2
-    - name: Install nightly
-      uses: actions-rs/toolchain@v1
+    - uses: actions/checkout@v4
+    - name: Downloads All Artifacts (root)
+      uses: actions/download-artifact@v4
       with:
-        toolchain: nightly
-    - name: Run benchmarks
-      run: benchmarks/run-all.sh
+        path: artifacts
+    - run: ls -al . artifacts/
+    - run: rustup update
+    - run: chmod +x artifacts/*/*.so
+    - name: Partition Benchmark Suite
+      run: |
+        ceil_div() { echo "($1 + $2 - 1) / $2" | bc; }
+        tmpdir=$(mktemp -d)
+        grep -v '^\#' benchmarks/all.suite > $tmpdir/all.suite
+        line_count=$(wc -l "$tmpdir/all.suite" | awk '{print $1}')
+        lines_per_chunk=$(ceil_div "$line_count" "$PARTITIONS")
+        split -d -l "$lines_per_chunk" "$tmpdir/all.suite" "${tmpdir}/all.split."
+        cp "$tmpdir/all.split.${{ matrix.partition }}" benchmarks/split.${{ matrix.partition }}.suite
+        rm -rf "$tmpdir"
```

Comment:
Seems like this parallelization logic could go in a script to be able to reproduce locally?

Reply:
Yeah, I can extract this into a script.  Apart from testing for CI changes, I don't think it would provide anything meaningful locally, but it could be useful for developers who need to make changes.